DataFormats/HcalRecHit/interface/HBHEChannelInfo.h

```diff
@@ -0,0 +1,213 @@
+        {return (rawCharge_[ts] - pedestal_[ts])*gain_[ts];}
+    inline double tsGain(const unsigned ts) const {return gain_[ts];}
+    inline uint8_t tsAdc(const unsigned ts) const {return adc_[ts];}
+    inline float tsRiseTime(const unsigned ts) const
+        {return hasTimeInfo_ ? riseTime_[ts] : HcalSpecialTimes::UNKNOWN_T_NOTDC;}
+
+    // Signal rise time measurement for the SOI, if available
+    inline float soiRiseTime() const
+    {
+        return (hasTimeInfo_ && soi_ < nSamples_) ?
+                riseTime_[soi_] : HcalSpecialTimes::UNKNOWN_T_NOTDC;
+    }
+
+    // The TS with the "end" index is not included in the window
+    inline double chargeInWindow(const unsigned begin, const unsigned end) const
+    {
+        double sum = 0.0;
+        const unsigned imax = end < nSamples_ ? end : nSamples_;
+        for (unsigned i=begin; i<imax; ++i)
+            sum += (rawCharge_[i] - pedestal_[i]);
+        return sum;
+    }
+
+    inline double energyInWindow(const unsigned begin, const unsigned end) const
+    {
+        double sum = 0.0;
+        const unsigned imax = end < nSamples_ ? end : nSamples_;
+        for (unsigned i=begin; i<imax; ++i)
+            sum += (rawCharge_[i] - pedestal_[i])*gain_[i];
+        return sum;
+    }
+
+    // The two following methods return MAXSAMPLES if the specified
+    // window does not overlap with the samples stored
+    inline unsigned peakChargeTS(const unsigned begin, const unsigned end) const
+    {
+        unsigned iPeak = MAXSAMPLES;
+        double dmax = -DBL_MAX;
+        const unsigned imax = end < nSamples_ ? end : nSamples_;
+        for (unsigned i=begin; i<imax; ++i)
+        {
+            const double q = rawCharge_[i] - pedestal_[i];
+            if (q > dmax)
+            {
+                dmax = q;
+                iPeak = i;
+            }
+        }
+        return iPeak;
+    }
+
+    inline unsigned peakEnergyTS(const unsigned begin, const unsigned end) const
+    {
+        unsigned iPeak = MAXSAMPLES;
+        double dmax = -DBL_MAX;
+        const unsigned imax = end < nSamples_ ? end : nSamples_;
+        for (unsigned i=begin; i<imax; ++i)
+        {
+            const double e = (rawCharge_[i] - pedestal_[i])*gain_[i];
+            if (e > dmax)
+            {
+                dmax = e;
+                iPeak = i;
+            }
+        }
+        return iPeak;
+    }
+
+    // The following function can be used, for example,
+    // in a check for presence of saturated ADC values
+    inline uint8_t peakAdcValue(const unsigned begin, const unsigned end) const
+    {
+        uint8_t peak = 0;
+        const unsigned imax = end < nSamples_ ? end : nSamples_;
+        for (unsigned i=begin; i<imax; ++i)
+            if (adc_[i] > peak)
+                peak = adc_[i];
+        return peak;
+    }
+
+private:
+    HcalDetId id_;
+
+    // Charge in fC for all time slices
+    double rawCharge_[MAXSAMPLES];
```

Comment:
Not the analyzers fault: your interface lets the outside to see only the bare pointer.
Within the scope of this class itself if you just use the bare array, the analyzer missing the issue can be blamed.

I thought of making this comment about std::array during review, but then (mistakenly?) assumed that the interfaces where this object is used force you to pass a bare pointer. In this case it didn't make much sense to switch to std::array just to lose the details in the scope where HBHEChannelInfo is used.